.github/workflows/ci.yml

```
@@ -152,19 +152,31 @@ jobs:
+  # on both. Each leg feeds GitHub's native code coverage (PR diff inline;
+  # requires Code Quality enabled in repo settings) and uploads its browsable
+  # HTML report as a downloadable CI artifact (no external hosting).
   coverage:
-    name: coverage
-    runs-on: ubuntu-latest
+    name: coverage (${{ matrix.tier }})
+    runs-on: ${{ matrix.os }}
     continue-on-error: true
     permissions:
       contents: read
       code-quality: write
```

actionlint (1.7.12)

[error] unknown permission scope "code-quality". all available permission scopes are "actions", "artifact-metadata", "attestations", "checks", "contents", "deployments", "discussions", "id-token", "issues", "models", "packages", "pages", "pull-requests", "repository-projects", "security-events", "statuses"

(permissions)